packages/labs/gen-wrapper-angular/src/index.ts

```diff
@@ -7,47 +7,144 @@
+import * as fs from 'fs/promises';
 import * as path from 'path';
+import {AbsolutePath} from '@lit-labs/analyzer/lib/paths.js';
+import {angularJsonProject} from './lib/angular-json-project.js';
+import {exec as execCb, ExecException} from 'child_process';
+import {promisify} from 'util';
+
+const exec = promisify(execCb);
+
+/**
+ * See ../../cli/src/lib/generate/generate.ts
+ */
+export const getCommand = () => {
+  return {
+    name: 'angular',
+    description: 'Generate Angular wrapper components from Lit elements',
+    kind: 'resolved',
+    async generate(options: GenerateOptions): Promise<FileTree> {
+      // TODO(justinfagnani): feed in ng generator specific command-line options
+      return await generateAngularWrapper(
+        options,
+        '' as unknown as AbsolutePath
+      );
+    },
+  };
+};
 
 export const generateAngularWrapper = async (
-  analysis: Package
+  options: GenerateOptions,
+  angularWorkspaceFolder: AbsolutePath
 ): Promise<FileTree> => {
+  console.log('Angular Generator');
+
+  const {analysis} = options;
+
   const litModules: LitModule[] = getLitModules(analysis);
-  if (litModules.length > 0) {
-    const packageName = analysis.packageJson.name;
-    if (packageName === undefined) {
-      throw new Error(
-        `Package must have a package name. Error in ${
-          analysis.rootDir + '/package.json'
-        }`
-      );
-    }
-    // TODO(justinfagnani): make configurable
-    const angularPackageName = `${packageName}-ng`;
-    // TODO(justinfagnani): put inside an Angular workspace
-    const angularPackageFolder = `${path.basename(analysis.rootDir)}-ng`;
-    return {
-      [angularPackageFolder]: {
-        '.gitignore': gitIgnoreTemplate(litModules),
-        'package.json': packageJsonTemplate(
-          angularPackageName,
-          analysis.packageJson,
-          litModules
-        ),
-        'tsconfig.json': tsconfigTemplate(),
-        ...wrapperFiles(analysis.packageJson, litModules),
-      },
-    };
-  } else {
+  if (litModules.length === 0) {
     throw new Error('No Lit components were found in this package.');
   }
+
+  const packageName = analysis.packageJson.name;
+  if (packageName === undefined) {
+    throw new Error(
+      `Package must have a package name. Error in ${
+        analysis.rootDir + '/package.json'
+      }`
+    );
+  }
+
+  try {
+    const {stdout: ngVersionOut} = await exec('ng --version');
+    console.log({ngVersionOut});
+  } catch (e: unknown) {
+    // See https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback
+    const err = e as ExecException & {stdout: string; stderr: string};
+    if (err.stderr.includes('command not found')) {
+      await installDepWithPermission({
+        description: 'The Angular CLI',
+        npmPackage: '@angular/cli',
+        global: true,
+        cwd: options.cwd,
+        stdin: options.stdin,
+        console: options.console,
+      });
+    }
```

Comment:
Should we check `ng --version` again afterwards, so we can give a clear error in case the install failed?